src/playlist-loader.js

```diff
@@ -432,7 +432,14 @@ const PlaylistLoader = function(srcUrl, hls, withCredentials) {
   /**
    * start loading of the playlist
    */
-  loader.load = () => {
+  loader.load = (isFinalRendition) => {
+    let refreshDelay;
+
+    if (isFinalRendition) {
+      refreshDelay = loader.media() ? loader.media().targetDuration * 1000 : 10 * 1000;
+      window.clearTimeout(mediaUpdateTimeout);
+      mediaUpdateTimeout = window.setTimeout(loader.load(false), refreshDelay);
+    }
```

Comment:
The `mediaUpdateTimeout` should be cleared immediately after this `if` block in case `load` is called from somewhere else and that caller doesn't want to wait the delay, we dont want the delayed call to run as well